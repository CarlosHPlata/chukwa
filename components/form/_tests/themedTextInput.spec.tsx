import { render, fireEvent } from "@testing-library/react-native";
import ThemedTextInput from "../themedTextInput";

describe("ThemedTextInput", () => {
  it ("renders correctly", () => {
    const tree = render(<ThemedTextInput label="Test Input" onChangeText={() => {}} />);

    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with initial value", () => {
    const { getByTestId } = render(
      <ThemedTextInput
        label="Test Input"
        value="Initial Value"
        onChangeText={() => {}}
      />
    );
    const input = getByTestId("themed-text-input:TextInput");
    expect(input.props.value).toBe("Initial Value");
  });

  it("calls onChangeText when text is changed", () => {
    const mockOnChangeText = jest.fn();
    const { getByTestId } = render(
      <ThemedTextInput
        label="Test Input"
        value=""
        onChangeText={mockOnChangeText}
      />
    );
    const input = getByTestId("themed-text-input:TextInput");
    
    fireEvent.changeText(input, "New Value");
    expect(mockOnChangeText).toHaveBeenCalledWith("New Value");
  });
});
