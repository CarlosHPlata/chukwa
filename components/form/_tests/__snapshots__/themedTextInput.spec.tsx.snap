// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedTextInput renders correctly 1`] = `
[
  <Text>
    Test Input
  </Text>,
  <TextInput
    inputMode="text"
    keyboardType="default"
    onChangeText={[Function]}
    placeholder="Test Input"
    style={
      {
        "borderWidth": 1,
        "marginBottom": 16,
        "padding": 8,
      }
    }
    testID="themed-text-input:TextInput"
    value=""
  />,
]
`;
